### DISCUSSION TITLE
Separate kotlin package into multiple packages

### DISCUSSION THREAD

**Discussion by satvik-s** [WarnerMedia/Rules-Engine#30] [Ideas]
The idea is to split up the kotlin implementation into two packages, `rulesengine-core` and `rulesengine-json-loader`.

Currently, jackson is the only dependency for the kotlin implementation. In order to make the library be fully and easily compatible with front-end clients, it might be better to split out the functionality to persist and store the JSON representation into a separate package and have to core package without any dependencies.

Credit for the idea to the person who approached at All Things Open to ask about cross-platform compatibility of the implementation.

Open to more feedback and thoughts about this!

P.S. use upvote if you align with the idea

**Comment by xsor-hbo** [comment]
This makes sense. The Kotlin package does not need to provide json loading out of the box.

**Comment by satvik-s** [comment]
Completed

https://github.com/WarnerMedia/Rules-Engine/pull/35
